providers/flagsmith/pkg/provider.go

```diff
@@ -0,0 +1,335 @@
+package flagsmith
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	flagsmithClient "github.com/Flagsmith/flagsmith-go-client/v2"
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+	"strconv"
+)
+
+type Provider struct {
+	client                  *flagsmithClient.Client
+	usingBooleanConfigValue bool
+}
+
+type ProviderOption func(*Provider)
+
+func NewProvider(client *flagsmithClient.Client, opts ...ProviderOption) *Provider {
+	provider := &Provider{
+		client: client,
+	}
+	for _, opt := range opts {
+		opt(provider)
+	}
+	return provider
+
+}
+
+// flagsmith provider does not have any hooks, returns empty slice
+func (p *Provider) Hooks() []of.Hook {
+	return []of.Hook{}
+}
+
+// Metadata returns value of Metadata (name of current service, exposed to openfeature sdk)
+func (provider *Provider) Metadata() of.Metadata {
+	return of.Metadata{
+		Name: "Flagsmith",
+	}
+}
+
+const TraitsKey = "traits"
+
+func (p *Provider) resolveFlag(ctx context.Context, flag string, defaultValue interface{}, evalCtx of.FlattenedContext) of.InterfaceResolutionDetail {
```

Comment:
Noticed that `ctx` is unused. It's up to you to decide to keep it or to remove it depending on your requirements

Reply:
Yeah, I think I would like to keep it(I have plans of updating our(flagsmith) go client to add support for this)